### PR TITLE
Use an interface method for IsDuplicate

### PR DESCRIPTION
**Pull request by tmthrgd**
This is simpler and more inline with other RR methods and eliminates the `isDuplicateRdata` switch statement. This also eliminates a panic that could occur if a record was lying about it's RRType (see `TestDuplicateWrongRrtype`).

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -81,6 +84,8 @@ func (h *RR_Header) len(off int, compression map[string]struct{}) int {
 	return l
 }
 
+func (r1 *RR_Header) isDuplicate(r2 RR) bool { return false }
```
Comment (miekg) [inline dns.go]: maybe nicer to implement the code that checks the header here instead of in the IsDuplicate (in duplicate.go)?

Comment (tmthrgd) [reply dns.go]: I had that before a3d7eba50825d546074d81084e639bbecf9cbb57, but changed it back to this approach. I’m happy to revert a3d7eba50825d546074d81084e639bbecf9cbb57 if you’d like.

Comment (tmthrgd) [reply dns.go]: The diff would like https://github.com/miekg/dns/compare/master...tmthrgd:a3d7eba%5E if I do revert that commit.

Comment (miekg) [reply dns.go]: why did you change it back? I think here is better, but would like to know the reason you reverted that?

Comment (tmthrgd) [reply dns.go]: @miekg Are you happy with how it is now?

**Pull request merged**